docs/service-deployment-concept.md

```diff
@@ -0,0 +1,105 @@
+# Service Deployment Concept
+
+This chapter outlines
+
+- the proposed structure for a service directory
+- how to use Helm charts and pipelines together
+- how to inject configuration into Helm charts
+- how to test Helm charts with pipeline dry-runs
+
+## Service artifacts
+
+All artifacts for a service deployment live in their own top-level directory in the ARO HCP repository and adheres roughly to the following structure:
+
+```plaintext
+my-service
+├── Makefile
+├── pipeline.yaml
+├── some-script.sh
+└── helm
+    └── <chart goes here>
+```
+
+### Helm chart
+
+Service and infrastructure deployments in MSFTs tenants need to be orchestrated by EV2. EV2 offers a various tooling types for deployment, with the most flexible being the [Shell](pipeline-concept.md#shell-step) extension. The tools that can be used within a Shell extension are well defined and offer `helm`. In addition, EV2 has support for direct Helm deployments in the working, which might be an option in the future.
```

Comment:
Worth linking ev2 docs listing image content?

Reply:
it's linked in the shell-step docs